haystack/components/routers/metadata_router.py

```diff
@@ -76,6 +76,11 @@ def __init__(self, rules: Dict[str, Dict]):
             ```
         """
         self.rules = rules
+        for rule in self.rules.values():
+            if "operator" not in rule:
+                raise ValueError(
+                    "Invalid filter syntax. See https://docs.haystack.deepset.ai/docs/metadata-filtering for details."
+                )
```

Comment:
I moved the check to the init method so a user is warned earlier if they have an invalid filter.